### PR TITLE
Align base

### PR DESCRIPTION
**Pull request by lyakh**
We're taking a new approach to this. Let's merge these 4 patches first, then we'll deal with the two more complex remaining patches from #3607 separately in PRs #3646 and #2647 and #3607 can be closed then.

### CODE REVIEW

**Comment by lyakh** [comment]
The first test was only modifying allocating from single buffers. It "only" failed 3 cases from the "Internal Intel CI System/merge/build" test suite. Next going to try only modifying contiguous buffer allocations.

**Comment by lyakh** [comment]
all tests passed except one - travis-ci/pr, which hasn't completed since almost 24 hours now. Restarting.

**Comment by lyakh** [comment]
SOFCI TEST

**Comment by lyakh** [comment]
This time travis-ci/pr completed successfully. This time several device tests failed, but since these failures weren't there last time, I think they're unrelated.

**Comment by lyakh** [comment]
@zrombel could you please have a look at the quickbuild failures here? Yesterday the test was green, no idea why it decided to re-test and why it's failing now.
@lgirdwood current test failures seem to be inconsistent, so, looks like this series is safe?

File: src/lib/alloc.c [hunk 1]
```diff
@@ -202,7 +202,7 @@ static void *align_ptr(struct mm_heap *heap, uint32_t alignment,
 
 /* allocate single block */
 static void *alloc_block(struct mm_heap *heap, int level,
-			 uint32_t caps, uint32_t alignment)
+			 uint32_t alignment)
```
Comment (lgirdwood) [inline src/lib/alloc.c:205]: ehm, they should be using this. Not all memory is equal. This fix should probably be a new PR.

**Review by lgirdwood — CHANGES_REQUESTED** [review]
This all good except path 3, which should be a separate fix in another PR·

File: src/lib/alloc.c (hunk 1 above)
Comment (lyakh) [reply src/lib/alloc.c:205]: ok, let's drop / postpone number 3 then, because they're interdependent

**Review by lgirdwood — APPROVED** [review]

**Comment by lyakh** [comment]
@lgirdwood interesting, this PR now shows a "request from change" from 2 hours ago from you and an approval from 24 minutes ago also from you... I'll dismiss your request for changes then

**Comment by lgirdwood** [comment]
Not sure why we see module unload fail here. rerun.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lyakh** [comment]
> Not sure why we see module unload fail here. rerun.

@lgirdwood failures are so strange. I'd bet they're unrelated, but I don't see many similar failures in other PRs around now. The failure on CNL seems again to be the old problem of IPC contents flipping to 0xffffffff: https://github.com/thesofproject/linux/issues/2498 , as for the other two failures on BSW and BDW I'd really need to debug them. I see a ton of "no bytes to consume" / "no bytes to produce" messages in logs and I don't see how this PR can introduce any functional changes. If anything it should only make alignment calculation slightly faster by avoiding one superfluous division - that's all! I'll split this into 3 PRs just to test...

**Comment by lgirdwood** [comment]
@lyakh CI looks good now - there has been a back log.

**Pull request merged**